core/frontend/src/render/webgl/glsl/AmbientOcclusion.ts

```diff
@@ -117,6 +117,9 @@ const computeAmbientOcclusion = `
     tOcclusion += curOcclusion;
   }
 
+  float distanceFadeFactor = pow(1.0 - clamp(nonLinearDepth / u_maxDistance, 0.0, 1.0), 2.0);
```

Comment:
I also wonder if we can just get rid of maxDistance if we are going to default it to 10km.

Reply:
I think that if your scaled sample size becomes less than a pixel, you've probably reached a reasonable max distance.

Reply:
What is the rationale behind this particular scaling formula?

Reply:
Based on OCP's approach - but they raised it to pow of 16, which was too aggressive for ours.

Reply:
I agree getting rid of maxDistance makes sense, though it worth noting that the scaled sample size is based on linearDepth, so as you rotate the camera, you will see some differences, but probably that's expected.  As it stands, you will see AO differences as you rotate and the texelStepSize shifts with the extents of the linearDepth.

Reply:
Either way it appears not to take the perspective frustum into account. Couldn't it make more sense to define your sample radius in meters and then convert it to texels in the shader using the perspective transform?

Reply:
That could be great if the core of the loop can work the same way.  Do you have a suggestion or a similar shader example in our code which takes a single value in meters and transforms it to texels?

Reply:
[modelToWindowCoordinates](https://github.com/iTwin/itwinjs-core/blob/6696fab86f6ffa7090c17b6f9bbee044819235f8/core/frontend/src/render/webgl/glsl/Viewport.ts#L31) does something similar.

Reply:
All right, tinkered a bit.

For our AO approach, I think the reason texelStepSize is preferable to something in meters is that the meaning of the meters step value would vary significantly among datasets, resulting in very thin or very large AO effects when applying a default number of meters.  Most people would be using defaults I imagine.  Using texels as the unit means the effect is always visible and a reasonable size on screen for a variety of geometry spans.

Regarding the distance cutoff, if we keep texels as the stepSize units, we cannot easily use "less than a pixel" applied stepSize for current position as a cutoff instead of maxDistance.  The user would have to set a large texelStepSize to affect the scene.  If a user sets texelStepSize of 1, nothing would render because the texelStepSize would immediately fade to 0 as it reaches toward the far plane.

In that case, keeping maxDistance makes sense -- but a question still remains if it should really still be applied if we will increase it so greatly anyway, since the existing 100 meters default was really low.

Reply:
@pmconne Old conversation, but I think it's resolved at least as far as this PR will take things -- could you mark resolved if agreed?